Amethyst/Managers/WindowManager.swift

```diff
@@ -21,13 +23,111 @@ enum WindowChange {
     case unknown
 }
 
-private struct ObserveApplicationNotifications {
+enum MouseState {
+    case pointing
+    case clicking
+    case dragging
+    case moving(window: SIWindow)
+    case resizing(screen: NSScreen, ratio: CGFloat)
+    case doneDragging(atTime: NSDate)
+}
+
+class MouseStateKeeper {
+    public var state: MouseState
+    private var _windowManager: WindowManager!
+    private var monitor: Any?
+
+    init() {
+        state = .pointing
+    }
+
+    var windowManager: WindowManager {
```

Comment:
I think this is a retain loop with the window manager, and deep linking to the window manager feels a bit odd. Possibly there should be a `MouseStateKeeperDelegate`?